src/pages/QuestionPage.tsx

```diff
@@ -9,56 +9,65 @@ import { useAppDispatch, useAppSelector } from "../state/reduxHooks";
 import { submitQuestion, setQuestionAnswer, submitQuestionExpired } from "../state/questionSlice";
 import { setGotQuestion, setRequestNextQuestion } from "../state/gameSlice";
 import { useSubmitAnswer } from "../hooks/useSubmitAnswer";
-import { useNextQuestion } from "../hooks/useNextQuestion"
+import { useNextQuestion } from "../hooks/useNextQuestion";
+import { setRequestUpdatedLeaderboard, resetLeaderboard } from "../state/leaderboardSlice";
+import { usePollForGetLeaderboard } from "../hooks/usePollForGetLeaderboard";
 import Timer from "../components/Timer";
+import Leaderboard from "../components/Leaderboard";
 
 export default function QuestionPage() {
 
   const { prompt, answers, index, correctAnswer } = useAppSelector(state => state.question);
   const { gameCode } = useAppSelector(state => state.game);
   const { isHost } = useAppSelector(state => state.player);
-  const { secondsLeft } = useAppSelector(state => state.time); 
+  const { secondsLeft } = useAppSelector(state => state.time);
   const [timeExpired, setTimeExpired] = useState<boolean>(false);
   const [answer, setAnswer] = useState<number | undefined>(undefined);
   const [answerArr, setAnswerArr] = useState<Array<string>>(["red", "blue", "green", "orange"]);
   const [showAnswerButtonClicked, setShowAnswerButtonClicked] = useState<boolean>(false);
   const [requestNextQuestionButtonPressed, setRequestNextQuestionButtonPressed] = useState<boolean>(false);
   const [answered, setAnswered] = useState<boolean>(false); // flag to indicate if user answered already or not
+  const [showLeaderboard, setShowLeaderboardButtonClicked] = useState<boolean>(false);
   const dispatch = useAppDispatch();
   const pollNextQuestionIntervalID = useRef<NodeJS.Timeout | undefined>(undefined);
 
   usePollForGetQuestion();
   useSubmitAnswer();
   useNextQuestion();
+  usePollForGetLeaderboard();
 
-  useEffect( () => {
+  useEffect(() => {
     // Resets flags on page once index changes
     console.log("resetting answer colors");
     setAnswerArr(["red", "blue", "green", "orange"]);
+    setShowLeaderboardButtonClicked(false);
     setShowAnswerButtonClicked(false);
     setTimeExpired(false);
     setAnswer(undefined);
     setRequestNextQuestionButtonPressed(false);
     // Stops polling for next question 
-    if (!isHost){
+    if (!isHost) {
       clearInterval(pollNextQuestionIntervalID.current);
     }
   }, [index])
 
-  useEffect( () => {
+  useEffect(() => {
     // sets timeExpired flag when secondsLeft reaches 0
     // otherwise sets the flag to false
-    if (secondsLeft === 0){
+    if (secondsLeft === 0) {
       setTimeExpired(true);
+      dispatch(resetLeaderboard());
+      dispatch(setRequestUpdatedLeaderboard(true));
```

Comment:
You're undoing what you did in [this line](https://github.com/Jhappy77/cpsc559-project-front-end/blob/76499c9ef64986de938fbd37f482f3f40383cc8a/src/state/leaderboardSlice.ts#L28) with this.